Fix Copy Password button when text is selected

c4rlo opened the pull request:
When the user chooses to copy the password for an entry to the clipboard, previously there was logic to check if text was selected, and if so, that text was instead copied to the clipboard. That made sense if (a) the user invoked the Copy Password action via its keyboard shortcut, and (b) that keyboard shortcut was configured (as per default) to be Ctrl-C, i.e. the same as the system action for copy-to-clipboard.

However, it made no sense if the user invoked that action in some other way, for example by clicking the corresponding toolbar button.

It also made no sense in the case that the Copy Password action had some other keyboard shortcut assigned. Also, if some other action had Ctrl-C assigned, the logic would not kick in then.

Fix all of the above by modifying the keyboard shortcut logic to intervene precisely in the case where a shortcut is pressed that matches the system copy-to-clipboard shortcut; only in that case do we now check if text is selected and if so copy that to the clipboard instead of the action we would otherwise take.

Fixes #10734.

## Testing strategy

Added test to `TestGui.cpp`. Also confirmed via manual testing.

## Type of change
- ✅ Bug fix (non-breaking change that fixes an issue)

droidmonkey commented:
Love it, glad to use the new action collection too! I really couldn't figure out a fix to this, too focused on the toolbar action and not the keyboard shortcut.

c4rlo commented on src/gui/MainWindow.cpp:
```diff
@@ -2166,33 +2073,94 @@ void MainWindow::initActionCollection()
-    ac->setDefaultShortcut(m_ui->actionLockDatabase, Qt::CTRL + Qt::Key_L);
-    ac->setDefaultShortcut(m_ui->actionLockAllDatabases, Qt::CTRL + Qt::SHIFT + Qt::Key_L);
-    ac->setDefaultShortcut(m_ui->actionEntryEdit, Qt::CTRL + Qt::Key_E);
-    ac->setDefaultShortcut(m_ui->actionEntryDelete, Qt::CTRL + Qt::Key_D);
-    ac->setDefaultShortcut(m_ui->actionEntryDelete, Qt::Key_Delete);
```
The second of these was overriding the first one, so only the Delete key was actually in effect. At this point I'm not sure what the desired behaviour is, so I've left this to only have the Delete key assigned to this action. I'm thinking this makes sense since the shortcut config UI only supports assigning a single shortcut to each action.

c4rlo commented:
I actually have a slightly simpler approach now, based on `QObject::installEventFilter()`. Moving this back to Draft while I consider it.

c4rlo committed:
Fix Copy Password button when text is selected

When the user chooses to copy the password for an entry to the
clipboard, previously there was logic to check if text was selected, and
if so, that text was instead copied to the clipboard. That made sense if
(a) the user invoked the Copy Password action via its keyboard shortcut,
and (b) that keyboard shortcut was configured (as per default) to be
Ctrl-C, i.e. the same as the system action for copy-to-clipboard.

However, it made no sense if the user invoked that action in some other
way, for example by clicking the corresponding toolbar button.

It also made no sense in the case that the Copy Password action had some
other keyboard shortcut assigned. Also, if some other action had Ctrl-C
assigned, the logic would not kick in then.

Fix all of the above by modifying the keyboard shortcut logic to
intervene precisely in the case where a shortcut is pressed that matches
the system copy-to-clipboard shortcut; only in that case do we now check
if text is selected and if so copy that to the clipboard instead of the
action we would otherwise take.

Add a test case to TestGui.cpp.

Fixes #10734.

c4rlo commented:
Ok, this is ready for review now. I've undone all the refactoring and kept this more focused.

c4rlo commented:
The macOS CI failure seems like some kind of test flakeyness, unrelated to this change. I have created #10901 to address that.
